scripts/codegen/options.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) 2018 Bobby Noelte.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import argparse
+
+from .cogapp.cogapp import CogOptions
+
+class Options(CogOptions):
+
+    code_start_marker = b'@code{.codegen}'
+    code_end_marker = b'@endcode{.codegen}'
+    code_insert_marker = b'@code{.codeins}@endcode'
+
+    @staticmethod
+    def is_valid_directory(parser, arg):
+        if not os.path.isdir(arg):
+            parser.error('The directory {} does not exist!'.format(arg))
+        else:
+            # File directory so return the directory
+            return arg
+
+    @staticmethod
+    def is_valid_file(parser, arg):
+        if not os.path.isfile(arg):
+            parser.error('The file {} does not exist!'.format(arg))
+        else:
+            # File exists so return the file
+            return arg
+
+    def __init__(self):
+        CogOptions.__init__(self)
+        self._parser = argparse.ArgumentParser(
```

Comment:
Are these options all required?

We don't use all these currently from what I see and since codegen is made for Zephyr specific usage I think we should try to limit its complexity to the use we have of it. This will ease maintenance.

I don't mean to limit codegen to pure dts instantiation generation, but I don't think neither zephyr should host a fully featured tool while it has a quite limited use of it. Besides, for introduction phase, we should rather limit ourselves to the basics and add bit  by bit what we need.

Reply:
Definitely option handling by codegen should adapted to the use in Zephyr. Finally CogOptions should be replaced by a minimal set of CodegenOptions.

>  Besides, for introduction phase

Yes - should be done after introduction.

Reply:
Point is I'm trying to minimize the code and complexity introduced at first.

Reply:
I will have a look. This stems from codegens basis Cog. I don´t want to introduce hidden problems into something that evolved over a long time.